Lib/logging/handlers.py

```diff
@@ -401,8 +402,12 @@ def doRollover(self):
                 timeTuple = time.localtime(t + addend)
         dfn = self.rotation_filename(self.baseFilename + "." +
                                      time.strftime(self.suffix, timeTuple))
-        if os.path.exists(dfn):
-            os.remove(dfn)
+        while os.path.exists(dfn):
+            if match := re.search(self.file_number, dfn):
+                name = re.sub(self.file_number, f'({int(match.groups()[1]) + 1})', dfn)
+                dfn = self.rotation_filename(name)
+            else:
+                dfn = self.rotation_filename(self.baseFilename + '.' + time.strftime(self.suffix, timeTuple) + ' (2)')
```

Comment:
Why the ` + ' (2)'` ?

Reply:
I was attempting to mimic the default behaviour I saw from Windows 10, in Windows if you create a new document  with a duplicate name it will create the documents like this:

```
New Document
New Document (2)
New Document (3)
```

On my Mac it omits the parenthesis and creating new documents yields a naming scheme like
```
New Document
New Document 2
New Document 3
```

I personally preferred the bracketed naming scheme but that is really just my personal preference. I don't know of anywhere else this pops up in the standard library but if there's a convention that I'm not following I'd be happy to rework that.